examples/website/wms/app.tsx

```diff
@@ -0,0 +1,160 @@
+// deck.gl, MIT license
+
+import React, {PureComponent} from 'react';
+import {render} from 'react-dom';
+import {StaticMap} from 'react-map-gl';
+
+import DeckGL from '@deck.gl/react/typed';
+import {MapController} from '@deck.gl/core/typed';
+import {ImageryLayer} from '@deck.gl/geo-layers/typed';
+import type {_ImageSourceMetadata as ImageSourceMetadata} from '@loaders.gl/wms';
+
+import ControlPanel from './components/control-panel';
+import {INITIAL_CATEGORY_NAME, INITIAL_EXAMPLE_NAME, INITIAL_MAP_STYLE, EXAMPLES} from './examples';
+
+export const INITIAL_VIEW_STATE = {
+  latitude: 49.254,
+  longitude: -123.13,
+  zoom: 11,
+  maxZoom: 16,
+  pitch: 45,
+  bearing: 0
+};
+
+export default class App extends PureComponent {
```

Comment:
Can we use function components?

If you split the example out to another PR I'm happy to help.

Reply:
Example broken out to separate PR: https://github.com/visgl/deck.gl/pull/7659

This example is from loaders.gl where being able to test different services is a goal. I agree that a deck.gl example could be much simpler.